### DISCUSSION TITLE
Question about how you are using TTS

### DISCUSSION THREAD

**Discussion by lordratner** [toverainc/willow-inference-server#88] [Q&A] [answered]
As I understand it, Willow does not yet respond verbally using the WIS TTS function, correct?

In that case, how exactly are people using the TTS functionality? I'd like to start playing around with it, but as far as I can tell the WebRTC only deals with ASR. 

I know these are basic questions, but I am a basic user 🤣😂.

I did finally get WIS working on Proxmox in an LXC, so I'm eager to explore.

**Comment by stintel** [comment, marked as answer]
Willow does support TTS using WIS: https://github.com/toverainc/willow/commit/db35f9bb1ea00b009aa1c17f8dab2afa7804b218

**Reply by lordratner** [reply]
Ok I'm an idiot. I had it all set up, it was just so quiet I didn't hear it, even though I set the volume to 50. I'll try maxing it out. 

Is there anyway to send specific text to a Willow to speak with TTS? Right now I only get the Home Assistant responses. This is just to play around with it mostly. 

Thanks!

**Reply by kristiankielhofner** [reply]
Hah, no worries! We actually have an internal note to normalize audio levels across the built in chimes, TTS output, etc because (as you've seen) you really have to crank up the volume to hear TTS responses. We've been tripped up by this ourselves!

WIS supports TTS with arbitrary text (of course) but as you've seen Willow just sends WIS TTS the text output as-is from Home Assistant. Currently the only way to change this is to [modify it in Home Assistant](https://github.com/home-assistant/intents/tree/main/responses).

Down the road we may look at some configuration, an architectural component, etc to enable Willow to send custom text for HA replies but this would be a pretty significant undertaking.
